src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -1873,9 +1912,45 @@ - (void)_performScheduledExpirationCheck
 
 - (BOOL)_attributeDataValue:(NSDictionary *)one isEqualToDataValue:(NSDictionary *)theOther
 {
-    // Attribute data-value dictionary should be all standard containers which
-    // means isEqual: comparisons all the way down, making a deep comparison.
-    return [one isEqualToDictionary:theOther];
+    // Attribute data-value dictionaries are equal if type and value are equal
```

Comment:
I know they weren’t here before, but the lack of nil checks, are anxiety inducing